### ISSUE TITLE
Builder warnings do not respect customizations

### ISSUE THREAD

**Issue by jfancher** [immutables/immutables#379]
If you have this:

````java
@Value.Immutable(builder = false)
abstract class Bug {
    static class Builder {
        public Builder(String foo) {}
   }
}
````

You get the warning:
````
Builder should have non-private no-argument constructor to be supported as Builder extend/super type
````

Despite the fact that no builder is being generated. The same is also true of the `Builder should be static and non-private` warning.

~~You also still get the warnings if you use `@Value.Style(typeBuilder = ...)` to change the name of the builder (and you _don't_ get the warning when there is an actual conflict with the customized name).~~ 
[edit: I think I was confused about this. The implementation class seems to always look for the name `Builder`, ignoring `typeBuilder=`, when looking to write the `extends`/`implements` declaration. That's either expected behavior or a separate issue.]

This seems like it could be a more general issue, but these are the specific cases I've run into. It's nothing critical since it's just a warning, but annoying nonetheless.

**Comment by elucash** [comment]
Refined validation for when the builder is disabled (will be released in the next release). IIRC `typeBuilder` configures pattern for a generated builder and `typeInnerBuilder` configures pattern to discover inner hand-written builder which can extend of be extended by generated builder.

**Comment by elucash** [comment]
Released in 2.2.9 Thank you!
